.github/workflows/integration-test.yml

```
@@ -42,3 +42,13 @@ jobs:
       - name: Run Integration Tests
         working-directory: .
         run: make integration-test
+      
+      - name: Create reports
+        working-directory: ./.github/workflows
+        run: ./create_reports.sh
+
+      - name: Upload cluster snapshot 
+        uses: actions/upload-artifact@v3
```

You need to add `if: always()` here to tell github actions to also run those steps when the previous tests failed
```suggestion
      - name: Create reports
        if: always()
        working-directory: ./.github/workflows
        run: ./create_reports.sh

      - name: Upload cluster snapshot
        if: always()
        uses: actions/upload-artifact@v3
```